### PR TITLE
Add customization to Chat util output

### PR DESCRIPTION
**Pull request by collindutter**
Addresses https://github.com/griptape-ai/griptape/issues/66.

Enables use cases like:
```python
import logging
from griptape.structures import Agent
from griptape.rules import Ruleset, Rule
from griptape.utils import Chat

pirate_agent = Agent(
    rulesets=[
        Ruleset(
            name="Pirate",
            rules=[
                Rule("You are a pirate"),
                Rule("You are worried about people stealing your gold"),
            ],
        )
    ]
)

Chat(
    pirate_agent,
    exit_keywords=["exit", "arr!"],
    processing_text="Arr, processing...",
    exiting_text="Arr, catch ye later...",
    intro_text="Arr - I be a pirate. Please chat with me!",
    prompt_prefix= "Hey: ",
    response_prefix= "Arr: ",
    output_fn=logging.error
).start()
```

```
ERROR:root:Arr - I be a pirate. Please chat with me!
Hey: Where is your treasure?
ERROR:root:Arr, processing...
[09/14/23 12:53:41] INFO     PromptTask 28d526294fbf4e54ac3c6e989a79c3ba
                             Input: Where is your treasure?
[09/14/23 12:53:48] INFO     PromptTask 28d526294fbf4e54ac3c6e989a79c3ba
                             Output: Arr matey, ye be a sly one, askin' about me treasure! I can't be revealin' the whereabouts of me precious gold, for fear of it bein'
                             plundered. Ye understand, don't ye? Always keepin' a weather eye open, I am.
ERROR:root:Arr: Arr matey, ye be a sly one, askin' about me treasure! I can't be revealin' the whereabouts of me precious gold, for fear of it bein' plundered. Ye understand, don't ye? Always keepin' a weather eye open, I am.
Hey: arr!
ERROR:root:Arr, catch ye later...

```

### CODE REVIEW

**Comment by shhlife** [comment]
Can we also provide prompt customization? So instead of Q: and A: we could have something like:

```
prompt_text = "Hey:",
response_text = "Arr:"
```

**Comment by collindutter** [comment]
@shhlife yep, updated!

File: griptape/utils/chat.py [hunk 1]
```diff
@@ -9,16 +9,24 @@
...
+    exit_keywords: list[str] = field(default=["exit"], kw_only=True)
+    exiting_text: str = field(default='exiting...', kw_only=True)
+    processing_text: str = field(default='processing...', kw_only=True)
+    intro_text: Optional[str] = field(default=None, kw_only=True)
+    prompt_prefix: str = field(default='Q: ', kw_only=True)
+    response_prefix: str = field(default='A: ', kw_only=True)
 
     def start(self) -> None:
+        if self.intro_text:
+            print(self.intro_text)
```
Comment (vasinov) [inline griptape/utils/chat.py:21]: Can we also make `print` customizable? May be something like this:

```python
output_message: Callable[[str], None] = field(
    default=Factory(lambda: print),
    kw_only=True
)
```

**Review by vasinov — CHANGES_REQUESTED** [review]

File: griptape/utils/chat.py [hunk 2]
```diff
@@ -9,16 +9,25 @@
 @define(frozen=True)
 class Chat:
     structure: Structure = field()
+    exit_keywords: list[str] = field(default=["exit"], kw_only=True)
+    exiting_text: str = field(default="exiting...", kw_only=True)
+    processing_text: str = field(default="processing...", kw_only=True)
+    intro_text: Optional[str] = field(default=None, kw_only=True)
+    prompt_prefix: str = field(default="Q: ", kw_only=True)
+    response_prefix: str = field(default="A: ", kw_only=True)
+    output_fn: Callable[[str], None] = field(default=Factory(lambda: print), kw_only=True)
```
Comment (vasinov) [inline griptape/utils/chat.py:18]: We use `func` in `SnowflakeSqlDriver.connection_func` and `fn` in `LocalVectorStoreDriver.relatedness_fn`. Some other callables look like regular methods (i.e., verb, no suffix).

Not sure what to do here. Just go with `fn` and address the rest later?

**Review by vasinov — APPROVED** [review]

**Pull request merged**